examples/common/convert_c4.py

```diff
@@ -110,21 +218,23 @@ class ConcatTokensC4(IterableDataset):
 
     def __init__(
         self,
+        dataset_name: str,
         split: str,
         tokenizer: PreTrainedTokenizerBase,
         max_length: int,
         bos_text: str,
         eos_text: str,
         no_wrap: bool,
+        data_subset: Union[str, None] = None
     ):
         self.tokenizer = tokenizer
         os.environ['TOKENIZERS_PARALLELISM'] = 'false'
         self.max_length = max_length
         self.bos_text = bos_text
         self.eos_text = eos_text
         self.should_wrap = not no_wrap
-        self.hf_dataset = hf_datasets.load_dataset(path='c4',
-                                                   name='en',
+        self.hf_dataset = hf_datasets.load_dataset(path=dataset_name,
```

Comment:
did you double check that the sharding produces different samples? we had checked c4, but i recall abhi being worried that the sharding might work differently for different datasets.

Reply:
I'm not sure what you mean!
Sharding of the newly-written dataset? Different how? Is this desirable or undesirable behavior?

Reply:
so in old version of hf datasets, the shards accessed would not be different per worker. so checking this would involve just running a dataloader with multiple workers and making sure each worker's first sample is different.

Reply:
I examined the first 5 batches w/ `num_workers=30` and the content of each batch was different, so I think we're good. Also happy to write up a more comprehensive test if you'd advise that.

Reply:
That is good enough for me for now. We're missing a lot of test coverage, and I think this is ok.